reconcile/aws_version_sync/merge_request_manager/merge_request.py

```diff
@@ -0,0 +1,189 @@
+RESOURCE_IDENTIFIER_REF = "resource_identifier"
+RESOURCE_ENGINE_REF = "resource_engine"
+RESOURCE_ENGINE_VERSION_REF = "resource_engine_version"
+
+COMPILED_REGEXES = {
+    i: re.compile(rf".*{i}: (.*)$", re.MULTILINE)
+    for i in [
+        VERSION_REF,
+        PROVIDER_REF,
+        ACCOUNT_ID_REF,
+        RESOURCE_PROVIDER_REF,
+        RESOURCE_IDENTIFIER_REF,
+        RESOURCE_ENGINE_REF,
+        RESOURCE_ENGINE_VERSION_REF,
+    ]
+}
+
+AVS_DESC = string.Template(
+    f"""
+This MR is triggered by app-interface's [aws-version-sync (AVS)](https://github.com/app-sre/qontract-reconcile/tree/master/reconcile/aws_version_sync).
+
+Please **do not remove the {AVS_LABEL} label** from this MR!
+
+Parts of this description are used by AVS to manage the MR.
+
+{PROMOTION_DATA_SEPARATOR}
+
+* {VERSION_REF}: {AVS_VERSION}
+* {PROVIDER_REF}: $provider
+* {ACCOUNT_ID_REF}: $account_id
+* {RESOURCE_PROVIDER_REF}: $resource_provider
+* {RESOURCE_IDENTIFIER_REF}: $resource_identifier
+* {RESOURCE_ENGINE_REF}: $resource_engine
+* {RESOURCE_ENGINE_VERSION_REF}: $resource_engine_version
+"""
+)
+
+
+class AVSInfo(BaseModel):
+    provider: str
+    account_id: str
+    resource_provider: str
+    resource_identifier: str
+    resource_engine: str
+    resource_engine_version: str
+
+
+class ParserError(Exception):
+    """Raised when some information cannot be found."""
+
+
+class ParserVersionError(Exception):
+    """Raised when the AVS version is outdated."""
+
+
+class Parser:
+    """This class is only concerned with parsing an MR deescription rendered by the Renderer."""
+
+    def _find_by_regex(self, pattern: re.Pattern, content: str) -> str:
+        if matches := pattern.search(content):
+            groups = matches.groups()
+            if len(groups) == 1:
+                return groups[0]
+
+        raise ParserError(f"Could not find {pattern} in MR description")
+
+    def _find_by_name(self, name: str, content: str) -> str:
+        return self._find_by_regex(COMPILED_REGEXES[name], content)
+
+    def parse(self, description: str) -> AVSInfo:
+        """Parse the description of an MR for AVS."""
+        parts = description.split(PROMOTION_DATA_SEPARATOR)
+        if not len(parts) == 2:
+            raise ParserError("Could not find data separator in MR description")
+
+        if AVS_VERSION != self._find_by_name(VERSION_REF, parts[1]):
+            raise ParserVersionError("Version is outdated")
+        return AVSInfo(
+            provider=self._find_by_name(PROVIDER_REF, parts[1]),
+            account_id=self._find_by_name(ACCOUNT_ID_REF, parts[1]),
+            resource_provider=self._find_by_name(RESOURCE_PROVIDER_REF, parts[1]),
+            resource_identifier=self._find_by_name(RESOURCE_IDENTIFIER_REF, parts[1]),
+            resource_engine=self._find_by_name(RESOURCE_ENGINE_REF, parts[1]),
+            resource_engine_version=self._find_by_name(
+                RESOURCE_ENGINE_VERSION_REF, parts[1]
+            ),
```

Comment:
I know this is a reuse for existing functions, another way to use less regex is to dump the whole metadata as a json/yaml code block, then it can be easily import/export using pydantic.

Reply:
This was my first thought as well, but this will have other limitations, e.g. future changes to the metadata may be more complicated. In this version, the MR description is more readable.